tests/SbrpTests/ValidationTests.cs

```diff
@@ -18,71 +18,69 @@
-
-    public ITestOutputHelper output { get; set; }
-
-    public ValidateSbrpTests(ITestOutputHelper _output)
+    public ValidationTests(ITestOutputHelper output)
     {
-        output = _output;
-
-        RepoRoot = Environment.CurrentDirectory.Substring(0, Environment.CurrentDirectory.IndexOf("artifacts"));
-
-        string buildPackagesDirectory = Path.Combine(RepoRoot, "artifacts/source-build/self/src/artifacts/packages/Release/Shipping");
+        Utilities.ValidateConfigParameters(new string[] { Config.RepoRootEnv, Config.BuildTypeEnv });
+    
+        Output = output;
+        string buildPackagesDirectory = Path.Combine(Config.RepoRoot, "artifacts/source-build/self/src/artifacts/packages", Config.BuildType, "Shipping");
 
         if (!Directory.Exists(buildPackagesDirectory))
         {
-            throw new DirectoryNotFoundException($"Directory {buildPackagesDirectory} does not exist. Try building with './build.sh -sb' first.");
+            throw new DirectoryNotFoundException($"Directory {buildPackagesDirectory} does not exist, try building with './build.sh'.");
         }
 
         Packages = Directory.GetFiles(buildPackagesDirectory, "*.nupkg", SearchOption.AllDirectories);
+
+        if (Packages.Length == 0)
+        {
+            throw new FileNotFoundException($"No packages found in {buildPackagesDirectory}");
+        }
     }
 
     [Fact]
     public void CheckForSbrpAttribute()
     {
-        string[] targetPacks = Directory.GetDirectories(Path.Combine(RepoRoot, "src/targetPacks/ILsrc")).Select(x => Path.GetFileName(x).ToLower()).ToArray();
-        string[] textOnlyPackages = Directory.GetDirectories(Path.Combine(RepoRoot, "src/textOnlyPackages/src")).Select(x => Path.GetFileName(x).ToLower()).ToArray();
-
-        var filteredPackages = new List<string>();
-        
-        foreach (var package in Packages)
-        {
-            string packageName = Path.GetFileNameWithoutExtension(package).ToLower();
-            packageName = Regex.Replace(packageName, VerionPattern, string.Empty);
-            if (!targetPacks.Contains(packageName) && !textOnlyPackages.Contains(packageName))
+        string[] targetPacks = Directory.GetDirectories(Path.Combine(Config.RepoRoot, "src/targetPacks/ILsrc"))
+            .Select(x => Path.GetFileName(x).ToLower())
+            .ToArray();
+        string[] textOnlyPackages = Directory.GetDirectories(Path.Combine(Config.RepoRoot, "src/textOnlyPackages/src"))
+            .Select(x => Path.GetFileName(x).ToLower())
+            .ToArray();
+
+        var filteredPackages = Packages
+            .Where(package =>
             {
-                filteredPackages.Add(package);
-            }
-        }
+                string packageName = Path.GetFileNameWithoutExtension(package).ToLower();
+                packageName = Regex.Replace(packageName, VersionPattern, string.Empty);
+                return !targetPacks.Contains(packageName) && !textOnlyPackages.Contains(packageName);
```

Comment:
It would probably be more efficient to use a `HashSet` for `targetPacks` and `textOnlyPackages` instead of array enumeration since there is going to be so many lookups.

Reply:
You're right. Not sure why that using a HashSet didn't cross my mind.  Added in https://github.com/dotnet/source-build-reference-packages/pull/817/commits/8ab54ea0adfc0dd81813d7d749c06f7eb416dbce